src/main/java/uk/gov/hmcts/reform/sscs/service/exceptions/FeignClientErrorDecoder.java

```diff
@@ -0,0 +1,89 @@
+
+    private final AppInsightsService appInsightsService;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    public FeignClientErrorDecoder(AppInsightsService appInsightsService) {
+        this.appInsightsService = appInsightsService;
+    }
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        Request originalRequest = response.request();
+        HttpMethod httpMethod = originalRequest.httpMethod();
+        HmcFailureMessage failMsg = null;
+
+        if (httpMethod.equals(HttpMethod.POST) || httpMethod.equals(HttpMethod.PUT)) {
+            HearingRequestPayload payload = mapToPostHearingRequest(originalRequest);
+            if (payload != null) {
+                failMsg = HmcFailureMessage.builder()
+                    .requestType(httpMethod.toString())
+                    .caseID(Long.valueOf(payload.getCaseDetails().getCaseRef()))
+                    .timeStamp(LocalDateTime.now())
+                    .errorCode(String.valueOf(response.status()))
+                    .errorMessage(response.reason())
+                    .build();
+            }
+        } else {
+            Long caseId = Long.parseLong(response.request().requestTemplate().queries().get("id").iterator().next());
+            failMsg = HmcFailureMessage.builder()
+                .requestType(httpMethod.toString())
+                .caseID(caseId)
+                .timeStamp(LocalDateTime.now())
+                .errorCode(String.valueOf(response.status()))
+                .errorMessage(response.reason())
+                .build();
+        }
+
+        switch (response.status()) {
+            case 400:
+            case 401:
+            case 403:
+            case 404: {
+                log.error("Error in calling Feign client. Status code "
+                              + response.status() + ", methodKey = " + methodKey);
+                log.error("Error details: {}", response.body().toString());
+                try {
+                    appInsightsService.sendAppInsightsEvent(failMsg);
+                } catch (JsonProcessingException e) {
+                    log.error("Error sending app insight event");
+                }
+                return new ResponseStatusException(HttpStatus.valueOf(response.status()),
+                                                   "Error in calling the client method:" + methodKey);
+            }
+            default:
+                return new Exception(response.reason());
+        }
+    }
+
+    private HearingRequestPayload mapToPostHearingRequest(Request request) {
+        HearingRequestPayload hearingRequestPayload = null;
+        try {
+            hearingRequestPayload = OBJECT_MAPPER.readValue(new String(request.body(), StandardCharsets.UTF_8),
+                                                            HearingRequestPayload.class);
+        } catch (JsonProcessingException e) {
```

Comment:
Can you throw the JsonProcessingException here after the log, also we should change the log message to something a bit more accurate + log the request too. I believe jackson will still process an empty request body. Have a look at the AppInsightsService#messageToJson method